### PR TITLE
fixes https://github.com/canvg/canvg/issues/655

### LINKED ISSUE #655: <animate /> tag breaking canvg when under <text /> node
First of all, thanks for this open source Library.

I noticed the `<animate />` breaks the canvg parser.
I had to replace the animate tag in my XML for it to work.
```
xml = new XMLSerializer().serializeToString(my-SVG-element);
xml = xml.replace(/<animate.*?\/>/g, '');
canvg(canvas, xml, { ignoreMouse: true, ignoreAnimation: true });
``` 

thanks.

### PR DESCRIPTION
**Pull request by gabelerner**
fixes #655

### CODE REVIEW

File: src/canvg.js [hunk 1]
```diff
@@ -2321,6 +2324,9 @@ function build(opts) {
 
     this.adjustChildCoordinates = function(ctx, textParent, parent, i) {
       var child = parent.children[i];
+      if (!(child instanceof svg.Element.TextElementBase)) {
+        return child;
```
Comment (gabelerner) [inline src/canvg.js]: prevents `<animate />` from running below code

File: src/canvg.js [hunk 2]
```diff
@@ -2148,6 +2148,9 @@ function build(opts) {
 
       // tween value linearly
       var newValue = p.from.numValue() + (p.to.numValue() - p.from.numValue()) * p.progress;
+      if (this.initialUnits === '%') {
+        newValue *= 100.0; // numValue() returns 0-1 whereas properties are 0-100
```
Comment (gabelerner) [inline src/canvg.js]: bug with %'s in animations

File: src/canvg.js (hunk 1 above)
Comment (brettz9) [reply src/canvg.js]: Could you perhaps duck-type here instead? Unless there is an ES6 `Symbol.hasInstance` (which does the duck-typing or what not), `instanceof` is not cross-frame safe...

File: src/canvg.js (hunk 2 above)
Comment (brettz9) [reply src/canvg.js]: LGTM... (Don't know if you want me to "resolve conversation")

File: src/canvg.js (hunk 1 above)
Comment (gabelerner) [reply src/canvg.js]: by `duck-type` you want `if (typeof(child.measureText) !== 'function')` or something similar?

Comment (brettz9) [reply src/canvg.js]: Yeah, exactly... If that method (or whatever combo of method/property checks can pin it down) is unique to that type...

Comment (gabelerner) [reply src/canvg.js]: done

**Pull request merged**